# 00 - Quick Inspection Dataset FAR-Trans

**Tujuan:** Memastikan semua data bisa di-load dengan benar dan melakukan inspeksi awal.

**Penulis:** Naufal Rizki Abyan (23082010235)  
**Tanggal:** 2026

---

## Notebook ini melakukan:
1. Import `load_data.py` dari folder `src/data/`
2. Load 4 file dataset FAR-Trans
3. Inspeksi singkat: shape, dtypes, head, missing values
4. Validasi bahwa loading bekerja konsisten

**Output yang diharapkan:** Konfirmasi bahwa semua data ter-load dengan tipe data yang benar dan siap untuk EDA mendalam di notebook berikutnya (`01_eda.ipynb`).

## 1. Setup Import Path

Karena notebook ada di folder `notebooks/` dan `load_data.py` di `src/data/`, kita perlu tambahkan project root ke Python path supaya bisa import.

In [3]:
import sys
from pathlib import Path

# Tambahkan project root ke Python path
project_root = Path().resolve().parent
sys.path.insert(0, str(project_root))

print(f"Project root: {project_root}")
print(f"Python path updated: ✓")

Project root: C:\All Projects\Transparent Investment Intentions Dashboard
Python path updated: ✓


## 2. Import Library & Load Data

In [4]:
import pandas as pd
from src.data.load_data import load_all

# Tampilkan lebih banyak kolom
pd.set_option('display.max_columns', 50)
pd.set_option('display.width', 200)

In [5]:
# Load semua data
data = load_all(stocks_only=False, verbose=True)

# Unpack ke variabel terpisah supaya gampang dipakai
customers = data['customers']
assets = data['assets']
transactions = data['transactions']
prices = data['prices']

Loading FAR-Trans Dataset
[OK] Customers loaded: 32,468 rows x 6 cols
  - Unique customers: 29,090
  - Duplicates (multiple snapshots): 3,378
[OK] Assets loaded: 836 rows x 9 cols
  - Unique ISINs: 806
  - Categories: {'MTF': 318, 'Stock': 293, 'Bond': 225}
[OK] Transactions loaded: 388,048 rows x 9 cols
  - Unique customers: 29,090
  - Unique assets: 320
  - Date range: 2018-01-02 00:00:00 to 2022-11-30 00:00:00
  - Buy/Sell: {'Buy': 228913, 'Sell': 159135}
[OK] Prices loaded: 703,303 rows x 3 cols
  - Unique ISINs: 807
  - Date range: 2018-01-01 00:00:00 to 2022-11-29 00:00:00
[OK] All data loaded successfully


## 3. Inspeksi: customer_information.csv

In [4]:
print("Shape:", customers.shape)
print("\nKolom & tipe data:")
customers.info()

Shape: (32468, 6)

Kolom & tipe data:
<class 'pandas.DataFrame'>
RangeIndex: 32468 entries, 0 to 32467
Data columns (total 6 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   customerID             32468 non-null  string        
 1   customerType           32468 non-null  category      
 2   riskLevel              32468 non-null  category      
 3   investmentCapacity     32468 non-null  category      
 4   lastQuestionnaireDate  32468 non-null  datetime64[us]
 5   timestamp              32468 non-null  datetime64[us]
dtypes: category(3), datetime64[us](2), string(1)
memory usage: 857.2 KB


In [5]:
customers.head(10)

,customerID,customerType,riskLevel,investmentCapacity,lastQuestionnaireDate,timestamp
0,DED5BF19E23CCCFEE322,Premium,Balanced,CAP_80K_300K,2021-11-30,2021-03-19
1,DED5BF19E23CCCFEE322,Premium,Balanced,CAP_80K_300K,2021-11-30,2022-01-21
2,6C0C752E66D5F0486C71,Mass,Income,Predicted_CAP_LT30K,2015-04-27,2018-01-02
3,3E8D8E4E07CD3210F25F,Mass,Income,CAP_30K_80K,2021-11-03,2018-01-02
4,37F996EE2A69B5C15DF3,Mass,Predicted_Conservative,Predicted_CAP_LT30K,2000-01-01,2018-01-02
5,062D57DE7F5817E2F5C3,Premium,Income,CAP_80K_300K,2020-07-16,2021-07-16
6,F2D15B652031EDFBEB7F,Mass,Conservative,CAP_LT30K,2020-04-01,2018-01-02
7,C7E103D420671A6774E3,Mass,Income,CAP_LT30K,2019-07-03,2020-01-16
8,59F6B634BD517563AA58,Mass,Predicted_Conservative,Predicted_CAP_LT30K,2000-01-01,2018-01-02
9,BBFE3AD3E027327D6F09,Mass,Predicted_Balanced,Predicted_CAP_80K_300K,2000-01-01,2020-03-12


In [6]:
# Distribusi target variable (riskLevel)
print("Distribusi Risk Level:")
print(customers['riskLevel'].value_counts())

Distribusi Risk Level:
riskLevel
Income                    10048
Balanced                   8684
Aggressive                 3202
Conservative               2916
Predicted_Conservative     2842
Predicted_Income           2704
Predicted_Balanced         1430
Not_Available               344
Predicted_Aggressive        298
Name: count, dtype: int64


In [7]:
# Cek missing values
print("Missing values:")
print(customers.isnull().sum())

Missing values:
customerID               0
customerType             0
riskLevel                0
investmentCapacity       0
lastQuestionnaireDate    0
timestamp                0
dtype: int64


## 4. Inspeksi: asset_information.csv

In [8]:
print("Shape:", assets.shape)
print("\nDistribusi assetCategory:")
print(assets['assetCategory'].value_counts())

assets.head()

Shape: (836, 9)

Distribusi assetCategory:
assetCategory
MTF      318
Stock    293
Bond     225
Name: count, dtype: int64


,ISIN,assetName,assetShortName,assetCategory,assetSubCategory,marketID,sector,industry,timestamp
0,GRF000011004,DHLOS PET OTE,DELPOIB,MTF,Balanced,AEDAK,<NA>,<NA>,2018-01-02
1,GRF000014008,DELOS FIXED INCOME PLUS - BOND FUND,DELDBDF,MTF,Structured,AEDAK,<NA>,<NA>,2018-01-02
2,GRF000022001,ΔΗΛΟΣ ΒΡΑΧΥΠΡΟΘΕΣΜΩΝ & ΜΕΣΟΠΡΟΘΕΣΜΩΝ ΕΠΕΝΔΥΣΕΩ...,DELPIMM,MTF,Money Market,AEDAK,<NA>,<NA>,2018-01-02
3,GRF000023009,DHLOS EISODHMATOS OMOLOGIAKO ESOTERIKOY,DELFXIB,MTF,Bond,AEDAK,<NA>,<NA>,2018-01-02
4,GRF000024007,DHLOS EUROBOND,DELEURB,MTF,Bond,AEDAK,<NA>,<NA>,2018-01-02


In [9]:
# Cek missing values - ini banyak yang missing!
print("Missing values di asset_information:")
print(assets.isnull().sum())
print(f"\nTotal rows: {len(assets):,}")

Missing values di asset_information:
ISIN                  0
assetName            24
assetShortName        1
assetCategory         0
assetSubCategory    269
marketID              0
sector              388
industry            439
timestamp             0
dtype: int64

Total rows: 836


In [10]:
# Khusus untuk saham (fokus penelitian kita)
stocks = assets[assets['assetCategory'] == 'Stock']
print(f"Total saham: {len(stocks):,}")
print(f"Unique ISIN saham: {stocks['ISIN'].nunique():,}")
print(f"\nDistribusi sektor (top 10):")
print(stocks['sector'].value_counts().head(10))

Total saham: 293
Unique ISIN saham: 285

Distribusi sektor (top 10):
sector
Industrials               44
Consumer Cyclical         41
Technology                30
Financial Services        26
Basic Materials           19
Consumer Defensive        18
Communication Services    13
Healthcare                12
Real Estate               12
Energy                    10
Name: count, dtype: Int64


## 5. Inspeksi: transactions.csv

In [11]:
print("Shape:", transactions.shape)
print("\nTipe data:")
transactions.info()

Shape: (388048, 9)

Tipe data:
<class 'pandas.DataFrame'>
RangeIndex: 388048 entries, 0 to 388047
Data columns (total 9 columns):
 #   Column           Non-Null Count   Dtype         
---  ------           --------------   -----         
 0   customerID       388048 non-null  string        
 1   ISIN             388048 non-null  string        
 2   transactionID    388048 non-null  int64         
 3   transactionType  388048 non-null  category      
 4   timestamp        388048 non-null  datetime64[us]
 5   totalValue       388048 non-null  float64       
 6   units            388048 non-null  float64       
 7   channel          388048 non-null  category      
 8   marketID         388048 non-null  string        
dtypes: category(2), datetime64[us](1), float64(2), int64(1), string(3)
memory usage: 21.5 MB


In [12]:
transactions.head()

,customerID,ISIN,transactionID,transactionType,timestamp,totalValue,units,channel,marketID
0,00017496858921195E5A,GRS434003000,7590224,Buy,2020-03-27,11000.0,5000.0,Internet Banking,XATH
1,00017496858921195E5A,GRS434003000,7607029,Sell,2020-04-06,12080.0,5000.0,Internet Banking,XATH
2,00017496858921195E5A,GRS434003000,7634872,Buy,2020-04-24,13400.0,5000.0,Internet Banking,XATH
3,00017496858921195E5A,GRS434003000,7652627,Sell,2020-05-07,12700.0,5000.0,Internet Banking,XATH
4,00017496858921195E5A,GRS434003000,7664807,Buy,2020-05-15,12150.0,5000.0,Internet Banking,XATH


In [13]:
# Statistik deskriptif kolom numerik
transactions[['totalValue', 'units']].describe()

,totalValue,units
count,3.880480e+05,3.880480e+05
mean,9.593622e+03,2.757254e+03
std,2.167012e+05,1.367003e+04
min,2.831069e-16,5.898060e-17
25%,4.810000e+02,1.000000e+02
50%,1.350000e+03,5.000000e+02
75%,3.664000e+03,1.570000e+03
max,2.715912e+07,2.150000e+06


In [14]:
# Distribusi channel transaksi
print("Distribusi channel:")
print(transactions['channel'].value_counts())

print("\nDistribusi Buy/Sell:")
print(transactions['transactionType'].value_counts())

Distribusi channel:
channel
Internet Banking    250805
Branch               71266
Phone Banking        65977
Name: count, dtype: int64

Distribusi Buy/Sell:
transactionType
Buy     228913
Sell    159135
Name: count, dtype: int64


## 6. Inspeksi: close_prices.csv

In [15]:
print("Shape:", prices.shape)
prices.head()

Shape: (703303, 3)


,ISIN,timestamp,closePrice
0,100974271034,2018-01-02,3.33
1,100974271034,2018-01-03,3.33
2,100974271034,2018-01-04,3.43
3,100974271034,2018-01-05,3.48
4,100974271034,2018-01-08,3.63


In [16]:
# Statistik harga
print("Statistik closePrice:")
print(prices['closePrice'].describe())

print(f"\nRange tanggal: {prices['timestamp'].min().date()} sampai {prices['timestamp'].max().date()}")

Statistik closePrice:
count    703303.000000
mean        103.588927
std         325.600827
min           0.010000
25%           4.049900
50%          29.332800
75%         100.111949
max        3783.500000
Name: closePrice, dtype: float64

Range tanggal: 2018-01-01 sampai 2022-11-29


## 7. Validasi Integritas Antar Tabel

Cek apakah ID yang ada di satu tabel juga ada di tabel lain (untuk memastikan join nantinya tidak menghasilkan banyak null).

In [17]:
# customerID di transactions vs di customers
customer_ids_trans = set(transactions['customerID'].unique())
customer_ids_master = set(customers['customerID'].unique())

print(f"Unique customerID di customers: {len(customer_ids_master):,}")
print(f"Unique customerID di transactions: {len(customer_ids_trans):,}")
print(f"customerID di transactions yang TIDAK ada di customers: {len(customer_ids_trans - customer_ids_master):,}")
print(f"customerID di customers yang TIDAK ada di transactions: {len(customer_ids_master - customer_ids_trans):,}")

Unique customerID di customers: 29,090
Unique customerID di transactions: 29,090
customerID di transactions yang TIDAK ada di customers: 0
customerID di customers yang TIDAK ada di transactions: 0


In [18]:
# ISIN di transactions vs di assets
isin_trans = set(transactions['ISIN'].unique())
isin_assets = set(assets['ISIN'].unique())

print(f"Unique ISIN di assets: {len(isin_assets):,}")
print(f"Unique ISIN di transactions: {len(isin_trans):,}")
print(f"ISIN di transactions yang TIDAK ada di assets: {len(isin_trans - isin_assets):,}")

Unique ISIN di assets: 806
Unique ISIN di transactions: 320
ISIN di transactions yang TIDAK ada di assets: 0


## 8. Kesimpulan Awal

Dari inspeksi awal, kita bisa identifikasi:

**Karakteristik Dataset:**
- ✅ 4 file ter-load dengan benar
- 📊 29.090 unique customers di customers table
- 📊 836 unique assets (293 Stock, 318 MTF, 225 Bond)
- 📊 388.048 transaksi dari 2018-2022
- 📊 ~700K rows harga harian untuk ~807 ISIN

**Hal Penting yang Harus Di-handle di Data Preparation:**
1. **Multiple snapshots di customers** — perlu deduplikasi (ambil yang terbaru per customerID)
2. **Banyak missing values di assets** — sector dan industry banyak null
3. **Risk level imbalanced** — distribusi tidak seimbang antar kelas
4. **Mix Predicted_* labels** — perlu dipisah dari label asli untuk training

**Next Step:** Lanjut ke `01_eda.ipynb` untuk Exploratory Data Analysis mendalam.